In [3]:
import pandas as pd
import numpy as np

# Problem 1:

In [11]:
bradyDataFrame = pd.DataFrame()
bradyDataFrame["Names"] = ["Greg", "Marcia", "Peter", "Jan", "Bobby", "Cindy", "Cousin Oliver"]
bradyDataFrame["Ages"] = [14, 12, 11, 10, 8, 6, 8]

print("Mean: " + str(bradyDataFrame["Ages"].mean()))
print("Median: " + str(bradyDataFrame["Ages"].median()))
print("Mode: " + str(bradyDataFrame["Ages"].mode()))

print("Variance: " + str(bradyDataFrame["Ages"].var()))
print("Standard Deviation: " + str(bradyDataFrame["Ages"].std()))
print("Standard Error: " + str(bradyDataFrame["Ages"].sem()))

Mean: 9.857142857142858
Median: 10.0
Mode: 0    8
dtype: int64
Variance: 7.476190476190475
Standard Deviation: 2.734262327610589
Standard Error: 1.0334540197243192


# Problem 2:

Since the data is symmetrical, I'm going to choose to use the mean as the estimate of the central tendency, and the standard deviation as the estimate of the variance.

# Problem 3:

In [12]:
bradyDataFrame.loc[bradyDataFrame["Names"] == "Cindy", "Ages"] += 1

print("Mean: " + str(bradyDataFrame["Ages"].mean()))
print("Median: " + str(bradyDataFrame["Ages"].median()))
print("Mode: " + str(bradyDataFrame["Ages"].mode()))

print("Variance: " + str(bradyDataFrame["Ages"].var()))
print("Standard Deviation: " + str(bradyDataFrame["Ages"].std()))
print("Standard Error: " + str(bradyDataFrame["Ages"].sem()))

Mean: 10.0
Median: 10.0
Mode: 0    8
dtype: int64
Variance: 6.333333333333333
Standard Deviation: 2.516611478423583
Standard Error: 0.9511897312113418


All of the variance estimates changed. The mean also changed. The only two estimates that did not change were the median and the mode.

# Problem 4:

In [13]:
bradyDataFrame.loc[(bradyDataFrame["Names"] == "Cousin Oliver") & (bradyDataFrame["Ages"] == 8)] = [["Jessica", 1]]

print("Mean: " + str(bradyDataFrame["Ages"].mean()))
print("Median: " + str(bradyDataFrame["Ages"].median()))
print("Mode: " + str(bradyDataFrame["Ages"].mode()))

print("Variance: " + str(bradyDataFrame["Ages"].var()))
print("Standard Deviation: " + str(bradyDataFrame["Ages"].std()))
print("Standard Error: " + str(bradyDataFrame["Ages"].sem()))

Mean: 9.0
Median: 10.0
Mode: 0     1
1     7
2     8
3    10
4    11
5    12
6    14
dtype: int64
Variance: 18.0
Standard Deviation: 4.242640687119285
Standard Error: 1.6035674514745462


Since the data is no longer as symmetrical, I would use median as the estimate of central tendency. I would also use variance to describe the non-normal distribution.

# Problem 5:

In [14]:
percentageArray = np.array([20, 23, 17, 5])

print("Percentage Median: " + str(np.median(percentageArray)))
print("Percentage Mean: " + str(np.mean(percentageArray)))

Percentage Median: 18.5
Percentage Mean: 16.25


I would use the median to estimate that around 18.5% of adult Americans were fans of the Brady Bunch. I would use median over mean because the data was non-symmetrical. The 5% from a single magazine tanked the mean, while the median was able to have an estimate closer to the middle of the data.